taf/GitRepository.py

```diff
@@ -44,6 +57,30 @@ def checkout_branch(self, branch_name, create=False):
       else:
         raise(e)
 
+  def clone_or_pull_up_to_commit(self, commit):
+    if not self.is_git_repository():
+      self.init_repo()
+    self.fetch()
+    self.merge_commit(commit)
+
+  def clone(self, no_checkout=False):
```

Comment:
I think we shuld have another param (e.g. `recreate`) to _explicitly_ tell that directory at `self.repo_path` will be deleted and recreated.

Moreover, `AuthenticationRepo` is a subclass of `GitRepository` and we don't want to accidentally remove the existing repo on a file system (I know that it will be cloned only if it does not exist, but anyway).

I think `repo.clone(True)` to remove directory and clone repo again is little bit safer.

Reply:
I added ` shutil.rmtree(self.repo_path, True)` to ease development. It should definitely be removed before this is merged.